### PR TITLE
backport: Merge bitcoin#22622, 22780, 22797, 22755, 22501, 21331  

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @knst for review

File: test/functional/p2p_invalid_tx.py [hunk 1]
```diff
@@ -175,8 +175,8 @@ def test_orphan_tx_handling(self, base_tx, resolve_via_block):
             ]
         }
         # Transactions that do not end up in the mempool
```
Comment (UdjinM6) [inline test/functional/p2p_invalid_tx.py:177]: 22797:
```suggestion
        # Transactions that do not end up in the mempool:
```

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -1332,7 +1332,7 @@ static void MaybePushAddress(UniValue & entry, const CTxDestination &dest)
 /**
  * List transactions based on the given criteria.
  *
- * @param  pwallet        The wallet.
+ * @param  wallet         The wallet.
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:1335]: Good catch! 👍 `develop` is currently broken because of this https://gitlab.com/dashpay/dash/-/pipelines/1381488325

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK 29720540b349a456cb4b3811d8b7674880bac204

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta requesting for review

**Review by knst — APPROVED** [review]
utACK 29720540b349a456cb4b3811d8b7674880bac204

**Pull request merged**